lib/Test/WWW/Stub/Handler.pm

```diff
@@ -0,0 +1,71 @@
+package Test::WWW::Stub::Handler;
+
+use strict;
+use warnings;
+
+use Carp qw(confess);
+
+sub new {
+    my ($class, %args) = @_;
+    my $pattern = $args{pattern};
+    my $app     = $args{app};
+    return bless { pattern => $pattern, app => $app }, $class;
+}
+
+sub factory {
+    my ($class, $uri_or_re, $app_or_res) = @_;
+    my $app;
+    my $type = ref($app_or_res);
+    if ($type eq 'CODE') {
+        $app = $app_or_res;
+    } elsif ($type eq 'ARRAY') {
+        $app = sub { $app_or_res };
+    } else {
+        confess 'Handler MUST be a PSGI app or an ARRAY';
+    }
+    return $class->new(pattern => $uri_or_re, app => $app);
+}
+
+sub is_regexp {
+    my ($self) = @_;
+    return ref($self->{pattern}) eq 'Regexp';
+}
+
+sub is_str {
```

Comment:
OK, all the `Handler` should know is how it compared, but not what it is.

I'll remove `is_str` and `is_regexp` and add `is_static_pattern` that means `pattern` should be compared with the URI using `eq` operator or not.
Do you think about it?